### ISSUE TITLE
Windows 10 hangs on booting screen

### ISSUE THREAD

**Issue by marco-04** [utmapp/UTM#2546]
**Describe the issue**
When starting the Windows 10 ARM VM, it hangs on the booting screen, where the UTM logo is showed. I tried to create multiple VMs, but the same problem happens. Windows 10 ARM image is the 21354 build converted to qcow2

**Configuration**
* UTM Version: 2.1.0 beta
* OS Version: 11.3.1
* Intel or Apple Silicon? Apple Silicon

**Crash log**
If the app crashed, you need a crash log. To get your crash log, open Console.app, go to `Crash Reports`, and find the latest entry for either QEMU, QEMUHelper, or qemu-\*. Right click and choose `Reveal in Finder`. Attach the report here.

**Debug log**
[debug.log](https://github.com/utmapp/UTM/files/6499577/debug.log)

**Upload VM**
VM made following the gallery guide for Windows 10 ARM

**Comment by muratdk23** [comment]
Same problem.

**Comment by pitermach** [comment]
Confirmed here as well on the 11.4 Public Beta

**Comment by masonasons** [comment]
I have this problem as well on 11.4 Beta.

**Comment by conath** [comment]
Can you try installing Windows from ISO? You can get a Windows for ARM ISO via a UUP website or link from [this page](https://chrisp.cafe/UTM/UTM-Win10ARM.html)

Note that you need to be members of the Windows Insider Program to be allowed to test Windows for ARM at this time.

**Comment by muratdk23** [comment]
Installed success with this settings:
System:
Architecture: ARM64 (Aarch64)
System: virt-6
Memory: 4096 MB (or more, depending on your device)
Show Advanced Settings: On
CPU: cortex-a57
CPU Cores: 4 Cores
Force Multicore: On
JIT Cache: 512 MB (optional)
QEMU Machine Properties: "highmem=off,virtualization=on"

But it installs and runs very slow.

**Comment by conath** [comment]
Are you on Intel? ARM on Intel is always slow.

**Comment by muratdk23** [comment]
> Are you on Intel? ARM on Intel is always slow.

No, I'm on arm. Without this "virtualization=on" parameter wm doesn't install at all. But in any case after restart wm windows tries to repair and fail.

**Comment by mccoy88f** [comment]
Same problem for me.
Also mb pro silicon m1 - It works on previous version of app but don't work on leatest beta (also trying using qemu 5.2)

**Comment by mccoy88f** [comment]
Also tested with iso (instead hd file from microsoft) do not work!

**Comment by muratdk23** [comment]
Btw last version that works normal with windows 10 is 2.0.25.

**Comment by conath** [comment]
> Btw last version that works normal with windows 10 is 2.0.25. On other versions install going well, but after relaunching doesn't work.

Please post a screenshot of what it gets stuck on.

**Comment by conath** [comment]
> Same problem for me.
> Also mb pro silicon m1 - It works on previous version of app but don't work on leatest beta (also trying using qemu 5.2)
> 
> Using virtualization=on it boot (i use iso to install it)
> but it's very slow

@mccoy88f 
Please share your current VM config where it boots (even if slowly). The Q35 system is x86-based, so it can‘t work with the Windows for ARM ISO.

**Comment by muratdk23** [comment]
> > Btw last version that works normal with windows 10 is 2.0.25. On other versions install going well, but after relaunching doesn't work.
> 
> Please post a screenshot of what it gets stuck on.

On what version? 2.1.0?

**Comment by muratdk23** [comment]
> > Btw last version that works normal with windows 10 is 2.0.25. On other versions install going well, but after relaunching doesn't work.
> 
> Please post a screenshot of what it gets stuck on.

<img width="1012" alt="Снимок экрана 2021-05-21 в 11 18 00" src="https://user-images.githubusercontent.com/79699274/119105864-6e7ad780-ba26-11eb-9ce1-926b3394ceca.png">
<img width="912" alt="Снимок экрана 2021-05-21 в 11 18 29" src="https://user-images.githubusercontent.com/79699274/119105890-72a6f500-ba26-11eb-9521-78f08af5aca9.png">

**Comment by conath** [comment]
@muratdk23  Please try to switch the Virtual Machine System to QEMU 5.2 ARM Virtual Machine. If looks like you have already installed Windows, so the change to QEMU 6.0 might be incompatible unless you reinstall.

**Comment by marco-04** [comment]
My image was already used with ACVM, should I use a clean image and retry?

> @muratdk23 Please try to switch the Virtual Machine System to QEMU 5.2 ARM Virtual Machine. If looks like you have already installed Windows, so the change to QEMU 6.0 might be incompatible unless you reinstall.

**Comment by muratdk23** [comment]
> @muratdk23 Please try to switch the Virtual Machine System to QEMU 5.2 ARM Virtual Machine. If looks like you have already installed Windows, so the change to QEMU 6.0 might be incompatible unless you reinstall.

I just reinstalled UTM, created new VM and launched it. On qemu 5.2 the same result.

**Comment by muratdk23** [comment]
> @muratdk23 Please try to switch the Virtual Machine System to QEMU 5.2 ARM Virtual Machine. If looks like you have already installed Windows, so the change to QEMU 6.0 might be incompatible unless you reinstall.

I just reinstalled UTM, created new VM and launched it. On qemu 5.2 the same result.

**Comment by muratdk23** [comment]
I just reinstalled UTM, created new VM and launched it. On qemu 5.2 the same result.

**Comment by muratdk23** [comment]
qemu-aarch64-softmmu: -accel tcg,tb-size=1024: qemu_mprotect__osdep: mprotect failed: Permission denied
qemu-aarch64-softmmu: falling back to tcg

Problem in mrpotect

**Comment by marco-04** [comment]
> My image was already used with ACVM, should I use a clean image and retry?

Update: just downloaded a new image from Microsoft website, converted to qcow2, and still doesn't boot

**Comment by osy** [comment]
I am able to reproduce this issue on v2.1.0 beta on M1 Mac Mini.

**Comment by osy** [comment]
Found the problem. I forgot to include the hypervisor patches in the QEMU v6.0.0 fork 😅

**Comment by marco-04** [comment]
lol. Well, waiting for an update then

> Found the problem. I forgot to include the hypervisor patches in the QEMU v6.0.0 fork 😅

**Comment by BestOwl** [comment]
77d6ca1 still has this issue on iPad OS 14.5.1
Applying `virtualization=on` parameter can make it boot, but it's very slow

I made a [custom build](https://github.com/BestOwl/UTM/releases/tag/77d6ca1) based on 77d6ca1 because v2.1.1 will crash on iPad OS 14.5.1.

**Comment by Thomas20090425** [comment]
actrually mine got to the press anykey to boot from CD/DVD but after pressed a key it will stuck on the boot screen with the UTM logo on the screen forever then it just doesn't want to boot with out the loading screen.

IPad Air 3,iOS 14.2 jailbroken, latest version of UTM.![image](https://user-images.githubusercontent.com/68798885/126260416-ff4f8445-4274-403c-b51e-bb11afbea7f3.jpeg)

**Comment by ssorguc** [comment]
Is there a solution to this?

![image](https://user-images.githubusercontent.com/17412267/129484476-138c5b98-78f3-4f31-b019-c4f1b9034d23.png)

**Comment by conath** [comment]
> Is there a solution to this?
> 
> ![image](https://user-images.githubusercontent.com/17412267/129484476-138c5b98-78f3-4f31-b019-c4f1b9034d23.png)

This screen means you need to install Windows. Check out my detailed guide to learn how to successfully install it: [video guide](https://youtu.be/KsNoGPczavw)

**Comment by zeyugao** [comment]
> actrually mine got to the press anykey to boot from CD/DVD but after pressed a key it will stuck on the boot screen with the UTM logo on the screen forever then it just doesn't want to boot with out the loading screen.
> 
> IPad Air 3,iOS 14.2 jailbroken, latest version of UTM.![image](https://user-images.githubusercontent.com/68798885/126260416-ff4f8445-4274-403c-b51e-bb11afbea7f3.jpeg)

Did you figure out how to solve it on iPad? I am using the Windows 11 on arm downloaded from uupdump and it stuck like your screenshot.

**Comment by conath** [comment]
> > actrually mine got to the press anykey to boot from CD/DVD but after pressed a key it will stuck on the boot screen with the UTM logo on the screen forever then it just doesn't want to boot with out the loading screen.
> > IPad Air 3,iOS 14.2 jailbroken, latest version of UTM.![image](https://user-images.githubusercontent.com/68798885/126260416-ff4f8445-4274-403c-b51e-bb11afbea7f3.jpeg)
> 
> Did you figure out how to solve it on iPad? I am using the Windows 11 on arm downloaded from uupdump and it stuck like your screenshot.

Try adding `,virtualization=on` to your System => additional machine properties. So the full text would be `highmem=off,virtualization=on`.
![CFADD76A-ADF3-4328-9DBE-F5DB3493B23B](https://user-images.githubusercontent.com/12073163/131250253-644bf9ad-d4e1-45ff-aa62-7d769c5b3c5d.jpeg)

(not sure where I have this hint from, can’t find the reference right now.)

**Comment by zeyugao** [comment]
> > > actrually mine got to the press anykey to boot from CD/DVD but after pressed a key it will stuck on the boot screen with the UTM logo on the screen forever then it just doesn't want to boot with out the loading screen.
> > > IPad Air 3,iOS 14.2 jailbroken, latest version of UTM.![image](https://user-images.githubusercontent.com/68798885/126260416-ff4f8445-4274-403c-b51e-bb11afbea7f3.jpeg)
> > 
> > 
> > Did you figure out how to solve it on iPad? I am using the Windows 11 on arm downloaded from uupdump and it stuck like your screenshot.
> 
> Try adding `,virtualization=on` to your System => additional machine properties. So the full text would be `highmem=off,virtualization=on`.
> ![AA7E0D19-4224-4130-99EF-C1765D4A363D](https://user-images.githubusercontent.com/12073163/131250426-fcde6286-a536-4703-b68d-e520c391958d.jpeg)
> 
> (not sure where I have this hint from, can’t find the reference right now.)

Well, thanks, indeed it works. But I glace through the comments above, saying that `virtualization=on` will make it run slow

**Comment by conath** [comment]
@zeyugao UTM is always slow on iPad, even when you use an ARM system. The problem is that Apple does not allow the same level of hardware access like on macOS, so we cannot do hardware virtualization. Instead UTM is using complete system emulation (QEMU TCG). 

This virtualization parameter is only a trick to make Windows ARM work.

**Comment by pavlotsyhanok** [comment]
If your Windows VM on UTM isn’t booting and just shows a blank screen or doesn’t seem to load the installer, here’s the fix:

As soon as the VM window opens, immediately press any key (e.g., Enter).

The Windows installation ISO only gives a very short window to detect a key press (“Press any key to boot from CD or DVD…”). On Apple Silicon Macs, the display sometimes lags slightly, so by the time you see that message, it might already be too late.

Once I pressed Enter right after the VM window appeared, the installer booted normally.
No other settings needed to be changed, I kept all defaults.
